File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.cpp
class Solution {
public:
    int sum(int digit) {
        int sum = 0;
        while (digit) {
            sum += digit % 10;
            digit = digit / 10;
        }
        return sum;
    }
    int minElement(vector<int>& nums) {
        int minEle = INT_MAX;
        for (int i = 0; i < nums.size(); i++) {
            int res = sum(nums[i]);
            minEle = min(minEle, res);
        }
        return minEle;
    }
};

File: 3300-minimum-element-after-replacement-with-digit-sum/3300-minimum-element-after-replacement-with-digit-sum.java
class Solution {
    // Time complexity: O(N logM)  ,   N = number of elements in the array  ,  M = maximum value in the array
    // Space complexity : O(1) 
    
    public int minElement(int[] nums) {
        int minEle = Integer.MAX_VALUE;

        for(int i = 0 ; i < nums.length; i++) {

            int digit = nums[i];
            int sum = 0;

            while(digit >  0) {
                sum += digit % 10;
                digit /= 10;
            }
            
            minEle = Math.min(minEle, sum);
        }
        return minEle;
    }
}